static/app/components/nav/utils.tsx

```diff
@@ -0,0 +1,158 @@
+import type {HTMLProps} from 'react';
+import type {LocationDescriptor} from 'history';
+
+import {isItemActive} from 'sentry/components/sidebar/sidebarItem';
+import {SIDEBAR_NAVIGATION_SOURCE} from 'sentry/components/sidebar/utils';
+import type {FeatureDisabledHooks, HookWithoutPrefix} from 'sentry/types/hooks';
+import normalizeUrl from 'sentry/utils/url/normalizeUrl';
+import type {useLocation} from 'sentry/utils/useLocation';
+
+export const NAV_DIVIDER = Symbol('divider');
+
+interface FeatureCheck {
+  features: string | string[];
+  hook?: HookWithoutPrefix<keyof FeatureDisabledHooks>;
+}
+
+interface NavItem {
+  label: string;
+  check?: FeatureCheck;
+}
+
+export interface SidebarItem extends SidebarItemBase {
+  to: string;
+  submenu?: SubmenuItem[];
+}
+
+export interface SidebarItemBase extends NavItem {
+  icon: JSX.Element;
+}
+interface SidebarItemTopLevel extends SidebarItemBase {
+  to: string;
+  submenu?: never;
+}
+
+interface SidebarItemWithChildren extends SidebarItemBase {
+  submenu: (SubmenuItem | false)[];
+  to?: never;
+}
+
+export interface SubmenuItem extends NavItem {
+  to: string;
+}
+
+export type NavItemRaw =
+  | SidebarItemTopLevel
+  | SidebarItemWithChildren
+  | typeof NAV_DIVIDER
+  | undefined
+  | false;
+
+export function resolveSidebarItem(
+  item: SidebarItemTopLevel | SidebarItemWithChildren
+): SidebarItem {
+  if (item.submenu) {
+    const submenu = item.submenu.filter(subitem => !!subitem) as SubmenuItem[];
+    const [{to}] = submenu;
+    return {...item, submenu, to};
+  }
+  return item;
+}
+
+export type NavigationItemStatus = 'inactive' | 'active' | 'active-parent';
+
+export function getNavigationItemStatus(
+  item: SidebarItem | SubmenuItem,
+  location: ReturnType<typeof useLocation>
+): NavigationItemStatus {
+  if (item.to.includes('/issues/') && item.to.includes('query=')) {
+    if (location.search.includes('viewId') && item.label === 'All') return 'active';
+    return hasMatchingQueryParam({to: item.to, label: item.label}, location)
+      ? 'active'
+      : 'inactive';
+  }
+  const normalizedTo = normalizeUrl(item.to);
+  const normalizedCurrent = normalizeUrl(location.pathname);
+  if (normalizedTo === normalizedCurrent) return 'active';
+  if (isItemActive({to: item.to, label: item.label})) return 'active';
+  if (
+    'submenu' in item &&
+    item.submenu?.find(
+      subitem =>
+        typeof subitem === 'object' &&
+        getNavigationItemStatus(subitem, location) !== 'inactive'
+    ) !== undefined
```

Comment:
nit could be .some
```suggestion
    item.submenu?.some(
      subitem =>
        typeof subitem === 'object' &&
        getNavigationItemStatus(subitem, location) !== 'inactive'
    )
```

Reply:
Nice catch! Addressed as part of [92d70b5](https://github.com/getsentry/sentry/pull/77665/commits/92d70b59ab5a518fa704b5483e958590d2fbe50c).